### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by Botlhale-Setou**
**Learning Objectives Reached:**

- Install Ruby on Rails framework.
- Use Ruby Gems as a software packages system.
- Understand what ORM is.
- Use database migration files to maintain database schema.

### CODE REVIEW

File: db/migrate/20230404171053_create_likes.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :author, foreign_key: { to_table: 'users' }
```
Comment (youmari) [inline db/migrate/20230404171053_create_likes.rb]: - [x] it should only be for the posts that has a foreign key called `authro_id` but for likes and comments it should be `user_id` as in the diagram

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @Botlhale-Setou , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Great commit messages
✔️ PRO PR
✔️ No linter error 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230404171053_create_likes.rb (hunk 1 above)
Comment (Botlhale-Setou) [reply db/migrate/20230404171053_create_likes.rb]: Updated 👍

File: db/migrate/20230404170721_create_comments.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class CreateComments < ActiveRecord::Migration[7.0]
   def change
     create_table :comments do |t|
-      t.references :author, foreign_key: { to_table: 'users' }
+      t.references :users, null: false, foreign_key: true
```
Comment (youmari) [inline db/migrate/20230404170721_create_comments.rb]: - [x] nice you are very close but you should remove the `s` from `users` so the foreign_key will be `user_id` as in the diagram not `users_id`

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @Botlhale-Setou , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 
✔️ Clean code 
✔️ PRO PR
✔️ No linter error
## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: db/migrate/20230404170721_create_comments.rb (hunk 2 above)
Comment (Botlhale-Setou) [reply db/migrate/20230404170721_create_comments.rb]: Thank you for the heads up, sir 👍 

I've gone ahead and **also** renamed the `posts_id` column to `post_id`, along with the `users_id` to `user_id` as per instruction and the ERD diagram.

image

Please note that the ERD diagram depicts the `user_id` in both the likes and comments tables as "AuthorID."
I made the switch to `user_id` from `author_id` in these tables as a result of the first code review.

image

**Review by Reem-lab — APPROVED** [review]
# Hi @Botlhale-Setou,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯


## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- No Linters error 🤩🔥💯
- Great working on the functionality 🔥🔥




## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**